### PR TITLE
Replace FontAwesome with React-FontAwesome

### LINKED ISSUE #2684: Replace Fontawesome with React-Fontawesome
**What existing feature could be improved?**
Add `React-Fontawesome` for cleaner React style syntax and individual/smaller imports. https://www.npmjs.com/package/@fortawesome/react-fontawesome

### PR DESCRIPTION
**Pull request by edplato**
## Description

Fixes #2684  .

- [x] Replace current `FontAwesome` full import class based library with individual React component based imports using https://www.npmjs.com/package/@fortawesome/react-fontawesome
- [x] Switched over to use `.min.css` versions of a few `app.js` `css` imports where available.

## Further comments

To import an icon into a React component

```
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

<FontAwesomeIcon icon={faTimes} />
```

A file with DOM element SVG exports was created for non-React components found in the `map` area of the app. These can eventually be replaced with the new `<FontAwesomeIcon />` components upon being refactored to use React.

## Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/master/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview

### CODE REVIEW

**Comment by edplato** [comment]
Nice, I'll fix those E2E selectors. It went pretty smooth switching over, but the one place where I could see hitting a wall was the `map` section of the app that wasn't React. I went with some SVG string DOM element imports there, but worth it in the end for the rest of the app to use the new React version.

File: web/js/app.js [hunk 1]
```diff
@@ -26,16 +26,15 @@ import ErrorBoundary from './containers/error-boundary';
 import Debug from './components/util/debug';
 
 // Dependency CSS
-import '../../node_modules/bootstrap/dist/css/bootstrap.css';
-import '../../node_modules/jquery-ui-bundle/jquery-ui.structure.css';
-import '../../node_modules/jquery-ui-bundle/jquery-ui.theme.css';
+import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
+import '../../node_modules/jquery-ui-bundle/jquery-ui.structure.min.css';
+import '../../node_modules/jquery-ui-bundle/jquery-ui.theme.min.css';
 import '../../node_modules/icheck/skins/square/grey.css';
```
Comment (jasontk19) [inline web/js/app.js]: I wonder if we can also remove these `icheck` css imports, as well as remove this dependency from the `package.json`?  At a glance it doesn't look like we are using this anywhere anymore.

File: web/js/components/util/list.js [hunk 2]
```diff
@@ -2,6 +2,42 @@ import React from 'react';
...
+  faFlag,
+  faGift,
+  faRuler,
+  faRulerCombined,
+  faTrash,
+  faTruck
+} from '@fortawesome/free-solid-svg-icons';
+
+// icons used with List by passing string as prop iconClass
+const listIcons = {
```
Comment (jasontk19) [inline web/js/components/util/list.js]:
```suggestion
const listIcons = {
  faArrowCircleDown,
  faArrowCircleUp,
  faBolt,
  faCircle,
  faCode,
  faEnvelope,
  faExclamationCircle,
  faFile,
  faFlag,
  faGift,
  faRuler,
  faRulerCombined,
  faTrash,
  faTruck
};
```

File: web/js/components/util/list.js [hunk 3]
```diff
@@ -2,6 +2,42 @@ import React from 'react';
...
+  faEnvelope: faEnvelope,
+  faExclamationCircle: faExclamationCircle,
+  faFile: faFile,
+  faFlag: faFlag,
+  faGift: faGift,
+  faRuler: faRuler,
+  faRulerCombined: faRulerCombined,
+  faTrash: faTrash,
+  faTruck: faTruck
+};
```
Comment (jasontk19) [inline web/js/components/util/list.js]: Not necessary but, I find simplified ES6 object properties a little less verbose and easier to read.

```suggestion
const listIcons = {
  faArrowCircleDown,
  faArrowCircleUp,
  faBolt,
  faCircle,
  faCode,
  faEnvelope,
  faExclamationCircle,
  faFile,
  faFlag,
  faGift,
  faRuler,
  faRulerCombined,
  faTrash,
  faTruck
};
```

File: web/js/app.js (hunk 1 above)
Comment (edplato) [reply web/js/app.js]: That would be good to remove. I'll check that out.

**Pull request merged**